src/devices/machine/e05a30.cpp

```diff
@@ -197,6 +205,10 @@ uint8_t e05a30_device::read(offs_t offset)
 	LOG("%s: e05a30_r([0xC0%02x]): ", machine().describe_context(), offset);
 
 	switch (offset) {
+	case 0x0:
+		m_c000_shift_register <<= 1;
+		result = (m_c000_shift_register & 0x1000000) ? 0x80 : 0x0;
```

Comment:
IMO this would be easier to read as `BIT(m_c000_shift_register, 24) << 7` – it makes it obvious you’re trying extract bit 24 of the value into bit 7 of the result.